### ISSUE TITLE
FS 9.12.12 shared memory date overflow bug?

### ISSUE THREAD

**Issue by varenius** [nvi-inc/fs#68]
Onsala has two VGOS telescopes; Ow and Oe. They are currently controlled by two FS 9.12.12 computers; freja and fulla. In recent weeks, a very strange FS bug appears to have hit both systems. 

freja first: On 2020-08-25 (i.e. day 238) freja FS log showed wrong timestamps like "2019.106.08:05:43.24", which means a shift back in time of (365-106)+238=497 days. 

fulla then: Today, 2020-09-14 (i.e. day 258), same problem. FS log showed "2019.126.04:08:16.05". This means a shift back in time of (365-126)+258=497 days. 

Because the problems were discovered when preparing for observations, the priority was to get FS back in operation rather than figuring out the root cause. In both cases, system time (e.g. "date" in terminal) was OK, only FS time was wrong. FS restart did not help, same problem. Therefore, I figured this has to do with some problem in FS shared memory which could be solved by a reboot. Indeed, after reboot, all seems well again.

Since the jump back in time was the same for both computers, on different dates, I presume this has something to do with system uptime and FS buffers. Unfortunately we did not log "uptime" of system before the reboot, and the computers had been up for so long (>45 days; likely many months) that the wtmp logs had been rotated more than once. I do not know of a way to figure out the time between the last two reboots with this long timescale. (Neither "last reboot" nor "grep reboot" in /var/log yielded anything useful). 

Perhaps this has already been fixed. I don't know if this is still a problem for FS10, but I thought better make a note of this in case it happens again.

**Comment by wehimwich** [comment]
I am sorry about this. I believe it is probably #56 rearing its head again, on the VGOS branch this time. It has all the markings of that issue. If it is that problem, the fix (the error message was missing) is already merged into FS10. That should give warnings of the problem before it occurs and notice once it does. I don't think this warrants a maintenance release of pre-FS10 versions.

I guess I am such an old timer that I don't expect computers to run for hundreds of days without being rebooted. :) The FSLx maintenance approach has monthly back-ups and weekly patching, all of which would make it unlikely that the computer would run longer than a few weeks without rebooting. More than 248 days without a reboot seems a little extreme to me in any event, but it is apparently more common now. :)

If you haven't already, I would recommend that you implement the `check_ntp` procedure as described in `/usr2/fs /misc/ntp.txt`, which also has suggestions for robust NTP usage. The most up-to-date version of the file is in FS10, e.g.,  at [ntp.txt](https://github.com/nvi-inc/fs/blob/master/misc/ntp.txt). Following the `check_ntp` approach described will also provide more diagnostic information in case of a problem. Please note that is important to make sure NTP has sync'd after a reboot before starting operations.

Please feel free to follow-up if you have further comments/questions.
